src/common/libflux/fripp.c

```diff
@@ -0,0 +1,456 @@
+/************************************************************\
+ * Copyright 2021 Lawrence Livermore National Security, LLC
+ * (c.f. AUTHORS, NOTICE.LLNS, COPYING)
+ *
+ * This file is part of the Flux resource manager framework.
+ * For details, see https://github.com/flux-framework.
+ *
+ * SPDX-License-Identifier: LGPL-3.0
+\************************************************************/
+
+#if HAVE_CONFIG_H
+#include "config.h"
+#endif
+
+#include <arpa/inet.h>
+#include <netinet/in.h>
+#include <stdbool.h>
+#include <sys/socket.h>
+#include <sys/stat.h>
+#include <syslog.h>
+#include <time.h>
+#include <unistd.h>
+
+#include "attr.h"
+#include "flog.h"
+#include "fripp.h"
+#include "reactor.h"
+
+#include "src/common/libczmqcontainers/czmq_containers.h"
+#include "src/common/libutil/iterators.h"
+
+#define FRIPP_MAX_PACKET_LEN 1440
+#define INTERNAL_BUFFSIZE 128
+#define DEFAULT_AGG_PERIOD 1.0
+
+typedef enum {
+    BRUBECK_COUNTER,
+    BRUBECK_GAUGE,
+    BRUBECK_TIMER
+} metric_type;
+
+union val {
+    ssize_t l;
+    double d;
+};
+
+struct metric {
+    union val cur;
+    union val prev;
+    bool inc;
+    metric_type type;
+};
+
+struct fripp_ctx {
+    struct sockaddr_in si_server;
+    int sock;
+    int buf_len;
+    int tail;
+    char *buf;
+    char prefix[INTERNAL_BUFFSIZE];
+
+    zhashx_t *metrics;
+    zlist_t *done;
+    flux_watcher_t *w;
+    double period;
+};
+
+bool fripp_enabled (flux_t *h, const char *metric)
```

Comment:
Why does this function take a `flux_t` handle?

Reply:
It takes a `flux_t` handle instead of a `fripp_ctx` to avoid trying to get/create a `fripp_ctx` each time it checks if collection is on.

Reply:
Maybe instead it should take a`fripp_ctx` and then if the `fripp_ctx` is `NULL` it can check if it's enabled on the instance.

Reply:
That function isn't using the `flux_t` though:
```c
bool fripp_enabled (flux_t *h, const char *metric)
{
    return getenv ("FLUX_FRIPP_STATSD") != NULL;
}
```